### PR TITLE
nss: Refactor namespace scheduler e2e to use new fixture

### PR DESCRIPTION
**Pull request by marun**

### CODE REVIEW

File: test/e2e/reconciler/namespace/controller_test.go [hunk 1]
```diff
@@ -39,126 +41,93 @@ const clusterLabel = "kcp.dev/cluster"
...
-			t.Errorf("incorrect number of servers: %d", len(servers))
-			return
-		}
-		server := servers[serverName]
-		cfg, err := server.Config()
-		if err != nil {
-			t.Error(err)
-			return
-		}
+	require.Equalf(t, len(f.Servers), 1, "incorrect number of servers")
```
Comment (stevekuznetsov) [inline test/e2e/reconciler/namespace/controller_test.go:64]: We've got a couple competing ideas about what to do here:

 - I use the Go standard library
 - you use testify
 - @davidfestal used some assertion library

Maybe let's find a common path and not simply introduce extra ones...?

Comment (imjasonh) [reply test/e2e/reconciler/namespace/controller_test.go:64]: FWIW, my personal vote is the Go standard library.

I find `require.Equal` is too easy to misuse, swapping `expected` and `actual` (as here).

Comment (marun) [reply test/e2e/reconciler/namespace/controller_test.go:64]: I'm using the same library as @davidfestal - it's testify in both cases.  And testify is a superset of what the go standard library does. Anyone that wants to laboriously check error codes and raise assertions is free to do so, arguably at the expense of readability.

Comment (ncdc) [reply test/e2e/reconciler/namespace/controller_test.go:64]: But we should agree as a community on a standard

**Comment by marun** [comment]
Added back `t.Parallel` to ensure the single test can run in parallel with other packages.

**Comment by marun** [comment]
Supplanted by #514, which updates all tests in a single PR.

**Pull request closed without merging**